### PR TITLE
Cleanup logic, remove unnecessary allocations in Request API.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13679)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
I'm curious! Does `normalize_method` exist only because we should stick to the spec? Why can't we directly turn it to a hyper method without upper-casing?

**Comment by frewsxcv** [comment]
@Wafflespeanut Good point. Will pivot this PR to clean that up.

File: components/script/dom/request.rs [hunk 1]
```diff
@@ -295,10 +296,9 @@ impl Request {
                 Some(s) => s,
                 None => return Err(Error::Type("Method is not a valid UTF8".to_string())),
             };
-            let normalized_method = normalize_method(method_string);
+            let method = normalize_method(method_string);
             // Step 25.3
-            let hyper_method = normalized_method_to_typed_method(&normalized_method);
-            *request.method.borrow_mut() = hyper_method;
+            *request.method.borrow_mut() = method;
```
Comment (KiChjang) [inline components/script/dom/request.rs]: Is `method` used anywhere else in this scope? If not, we can just make the above two lines into one.

Comment (frewsxcv) [inline components/script/dom/request.rs]: I kept the separation because it better corresponds to the spec. Normalization is 25.2. Setting is 25.3.

Comment (KiChjang) [inline components/script/dom/request.rs]: Perhaps meld these two lines into the previous match, and annotate the Some branch with 25.3? I don't see the need to create 3 different local variables and ending up throwing them away afterwards.

Comment (frewsxcv) [inline components/script/dom/request.rs]: I'm not entirely sure what you're suggesting, but check out my latest force push.

File: components/script/dom/request.rs [hunk 2]
```diff
@@ -290,15 +291,12 @@ impl Request {
                 return Err(Error::Type("Method is forbidden".to_string()));
             }
             // Step 25.2
-            let method_lower = init_method.to_lower();
-            let method_string = match method_lower.as_str() {
-                Some(s) => s,
+            let method = match init_method.as_str() {
+                Some(s) => normalize_method(s),
```
Comment (KiChjang) [inline components/script/dom/request.rs]: I meant that add `Step 25.3` above this line, and make it `*request.method.borrow_mut() = normalize_method(s)` instead.

Comment (frewsxcv) [inline components/script/dom/request.rs]: I still don't understand. Can you paste exactly what you'd like?

Comment (KiChjang) [inline components/script/dom/request.rs]:
```rust
match init_method.as_str() {
    // Step 25.3
    Some(s) => *request.method.borrow_mut() = normalize_method(s),
    None => return Err(..)
}
```

Comment (frewsxcv) [inline components/script/dom/request.rs]: `normalize_method` is step 25.2.

**Comment by KiChjang** [comment]
On second thought, this doesn't do any heap allocation so the let-binding is fine.

@bors-servo r+

**Comment by KiChjang** [comment]
@bors-servo retry
* #13662

File: components/script/dom/request.rs [hunk 3]
```diff
@@ -458,28 +456,16 @@ fn net_request_from_global(global: &GlobalScope,
...
-        "POST" => HttpMethod::Post,
-        "PUT" => HttpMethod::Put,
-        a => HttpMethod::Extension(a.to_string())
-    }
-}
-
 // https://fetch.spec.whatwg.org/#concept-method-normalize
-fn normalize_method(m: &str) -> String {
+fn normalize_method(m: &str) -> HttpMethod {
     match m {
```
Comment (Ms2ger) [inline components/script/dom/request.rs]: `match_ignore_ascii_case` might have been better here.

**Pull request merged**